src/js/jsx/sections/transform/Rotate.jsx

```diff
@@ -74,14 +67,21 @@ define(function (require, exports, module) {
             var differentSource = curDocument.id !== nextDocument.id ||
                 !Immutable.is(curLayerIDs, nextLayerIDs);
 
-            // HACK: Now is the best place to update these values as they shouldn't affect render
-            // and we want to update them only when selection changes without re-render
+            // If selection has changed, reset the value back to zero
             if (differentSource) {
-                this._lastAngle = 0;
-                this._scrubAngle = 0;
+                this.setState({
+                    differentSource: true,
+                    value: 0
+                });
+            } else {
```

Comment:
I amended my original commit with this additional  `else` block - to avoid unnecessary renders